ContosoCrafts.Web.Server/Controllers/JsonFileProductService.cs

```diff
@@ -0,0 +1,144 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using ContosoCrafts.Web.Server.Models;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using Stripe.Checkout;
+
+namespace ContosoCrafts.Web.Server.Controllers
+{
+    /// <summary>
+    /// Json File Product Service.
+    /// </summary>
+    /// <seealso cref="ContosoCrafts.Web.Server.Controllers.IProductService" />
+    public class JsonFileProductService : IProductService
+    {
+        /// <summary>
+        /// The logger
+        /// </summary>
+        private readonly ILogger<JsonFileProductService> logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JsonFileProductService" /> class.
+        /// </summary>
+        /// <param name="logger">The logger.</param>
+        public JsonFileProductService(ILogger<JsonFileProductService> logger)
+        {
+            this.logger = logger;
+
+            var manifestEmbeddedProvider = new ManifestEmbeddedFileProvider(typeof(Program).Assembly);
+            var fileInfo = manifestEmbeddedProvider.GetFileInfo("_data/products.json");
+            using var reader = new StreamReader(fileInfo.CreateReadStream());
+
+
+            var fileContent = reader.ReadToEnd();
+
+            Products = JsonSerializer.Deserialize<List<Server.Models.Product>>(fileContent,
+                   new JsonSerializerOptions
+                   {
+                       PropertyNameCaseInsensitive = true
+                   });
+        }
+
+        /// <summary>
+        /// Gets the products.
+        /// </summary>
+        /// <value>
+        /// The products.
+        /// </value>
+        private List<Server.Models.Product> Products { get; }
+
+        /// <summary>
+        /// Gets the products.
+        /// </summary>
+        /// <returns></returns>
+        public Task<IEnumerable<Server.Models.Product>> GetProducts() => Task.FromResult(Products.AsEnumerable());
+
+        /// <summary>
+        /// Gets the product.
+        /// </summary>
+        /// <param name="productId">The product identifier.</param>
+        /// <returns></returns>
+        public Task<Server.Models.Product> GetProduct(string productId)
+        {
+            var selectedProduct = new Product();
+
+            foreach (var p in Products)
```

Comment:
For each loop is iterating over all items unnecessary, local variable selectedPizza not required, Product can be returned directly for matching Product.Id in foreach loop
- In fact loop is not required, we can use LINQ expression instead: `var product = Products.FirstOrDefault(p => p.Id == id);	`